src/states/elder/tests.rs

```diff
@@ -496,30 +457,6 @@ fn when_accumulate_offline_and_start_dkg_and_section_info_then_node_is_removed_f
     assert!(!elder_test.is_candidate_elder());
 }
 
-#[test]
-fn accept_node_before_and_after_reaching_elder_size() {
-    // Set section size to one less than the desired number of the elders in a section. This makes
-    // us reject any bootstrapping nodes.
-    let mut elder_test = ElderUnderTest::new(ELDER_SIZE - 1);
-    let node0 = JoiningNodeInfo::with_addr(&mut elder_test.rng, "198.51.100.0:5000");
-
-    // Bootstrap succeed even with too few elders.
-    elder_test.handle_bootstrap_request(*node0.public_id(), node0.connection_info());
-    assert!(elder_test.is_connected(&node0.connection_info().peer_addr));
-
-    let node1 = JoiningNodeInfo::with_addr(&mut elder_test.rng, "198.51.100.1:5000");
-
-    // Add new section member to reach elder_size.
-    let new_info = elder_test.new_elders_info_with_candidate();
-    elder_test.accumulate_online(elder_test.candidate.clone());
-    elder_test.accumulate_start_dkg(&new_info);
-    elder_test.accumulate_section_info_if_vote(&new_info);
-
-    // Bootstrap succeeds.
-    elder_test.handle_bootstrap_request(*node1.public_id(), node1.connection_info());
-    assert!(elder_test.is_connected(&node1.connection_info().peer_addr));
```

Comment:
While this test used connected to verify a property, I don't think it is what it test, it test that a node can join.
I'd be incline to think we should still have this test, just using a different property to validate it is the case.

BTW, we don't really have responsiveness check for Adults, how do we detect they drop with this PR?

Reply:
> we should still have this test, just using a different property to validate it is the case.

Maybe we can check that the elder sent `BootstrapResponse` back? Also, we don't do that thing where we reject the request if we don't have enough members anymore, so that part of the test is not necessary.

> BTW, we don't really have responsiveness check for Adults, how do we detect they drop with this PR?

We don't do drop detection by responsiveness yet. We rely on the `ConnectionFailure` event from quic-p2p. This has not changed in this PR.